### DISCUSSION TITLE
Unresponsive flight

### DISCUSSION THREAD

**Discussion by thijsvdc** [sanhozay/PilotLog#15] [General]
PilotLog version   :  2.0.0
Java version       :   15.0.2
Flightgear version :  2020.3.1
O/S                :  Debian LXC
O/S version        :  10
Browser            :  FF

## Problem Description

Hi,

First of all: thanks a lot for this great tool, I'm really loving it! The maps, altitude graph, airports visited,... great!

I was very dissapointed to see that my flight today was not logged. This was shown in the logs:
```
INFO 152 --- [nio-8080-exec-1] o.f.pilotlog.service.FlightService       : Ended flight #19 in A320neo-CFM from LFRS to LXGB, status INVALID
INFO 152 --- [   scheduling-1] o.f.pilotlog.service.FlightService       : Deleted unresponsive flight #19 in A320neo-CFM from LFRS to LXGB, status INVALID
```

What are possible reasons for PilotLog to see the flight as "unresponsive"? FYI, the complete flight was correctly logged to https://fgtracker.ml.

Thanks!

**Comment by sanhozay** [comment]
Great, glad you are finding it useful!

The message about deleting an unresponsive flight is slightly misleading. There is a periodic housekeeping process that removes any flights that have not been updated for 60s. The original purpose was to clean up a flight after a sim crash but I adapted it to clean up invalid flights too - I just never updated the log message.

The thing to focus on is that flight 19 had a status of INVALID when you landed. There should be an indication of how it became invalid higher up in the log file but possible reasons are pausing the flight, using time-warp or landing with more fuel than you started. There is a short grace period on the pause key 'p' because I kept hitting it accidentally using the Flightgear flaps up key '['. If you accidentally pause the sim but unpause again within 5s, the flight is not invalidated.

An invalid flight will be shown in red and strike-through in the flight record but only until you land. Once you land and stop updating the flight, the housekeeping process picks it up and deletes it.

**Reply by thijsvdc** [reply]
You nailed it. I accidentally activated time-warp for a couple of seconds, that must have been the reason.
Thanks for the detailed information!

**Reply by sanhozay** [reply]
OK, I'll add a task to modify the Nasal script so that accidental time warps are treated the same as pauses. I'll also make a suitable change to the misleading log message.
